app/utils/formats.py

```diff
@@ -213,6 +213,53 @@ def from_donation(cls, item: ANHash) -> Optional[ATRecord]:
             custom={},
         )
 
+    @classmethod 
+    def from_mobilize(cls, data: Dict[str, str]) -> ATRecord:
+        if (timeslot_id := data.get("timeslot id")) and (event_id := data.get("event id")):
+            shift_id = f"{data['email']}-{timeslot_id}-{event_id}"
+        else:
+            shift_id = f"{data['email']}-{data['signup created time']}"
```

Comment:
One small nit: I believe that event id may exist if there are no time slots (so-called "unshifted events").  So I recommend the following:
```suggestion
        if event_id := data.get("event id"):
            if timeslot_id := data.get("timeslot id"):
                shift_id = f"{data['email']}-{event_id}-{timeslot_id}"
            else:
                shift_id = f"{data['email']}-{event_id}-{data['signup created time']}"
        else:
            shift_id = f"{data['email']}-{data['signup created time']}"
```
That way you use all the info to identify the event you can.